### ISSUE TITLE
Pivo Banking app

### ISSUE THREAD

**Issue by voto123** [devadvance/rootcloak#97]
This is an nfc-payment -enabled banking app.

I have tried RootCloak 3.0 beta 20160731

When opening Pivo for the first time it seems to be force closing if xposed framework is installed, so I uninstalled framework, opened the app, installed framework back and then tried the latest rootcloak.
App still says "Access denied - Unfortunately cant use Pivo while your device is in developer mode"

Play Store link:
https://play.google.com/store/apps/details?id=fi.op.android.lompsa

If its somehow possible for me to find how this app detects root/xposed/developer mode, I will gladly help you as much as I can.

**Comment by pylerSM** [comment]
"if they have developer mode enabled."

gonna be faked too
https://github.com/devadvance/rootcloak/pull/99

**Comment by voto123** [comment]
@hikaritenchi Hi, I would like to reopen this issue. I updated the description a bit.
I can also create a new issue too if this doesnt get reopened in few days

**Comment by pylerSM** [comment]
So...

- Try to clear app data and close it
- RootCloak - Manage apps - Click + to add your app.
- Start your app

does not work for you? :D

**Comment by voto123** [comment]
@pylerSM Nope... Did all that. And double checked everything with no luck.

- App added on RootCloak
- All data cleared of Pivo + force stop
- Access still denied

**Comment by pylerSM** [comment]
can you enable Debug logs in rootcloak and rerun your app? :) and then get Xposed log and share it here? :D

**Comment by voto123** [comment]
07-31 14:09:39.902 I/Xposed  ( 7780): -----------------
    07-31 14:09:39.902 I/Xposed  ( 7780): Starting Xposed version 86, compiled for SDK 23
    07-31 14:09:39.902 I/Xposed  ( 7780): Device: A0001 (OnePlus), Android version 6.0.1 (SDK 23)
    07-31 14:09:39.902 I/Xposed  ( 7780): ROM: bacon-userdebug 6.0.1 MOB30R db45b6937d test-keys
    07-31 14:09:39.902 I/Xposed  ( 7780): Build fingerprint: oneplus/bacon/A0001:6.0.1/MHC19Q/ZNH2KAS1KN:user/release-keys
    07-31 14:09:39.902 I/Xposed  ( 7780): Platform: armeabi-v7a, 32-bit binary, system server: yes
    07-31 14:09:39.902 I/Xposed  ( 7780): SELinux enabled: yes, enforcing: no
    07-31 14:09:42.087 I/Xposed  ( 7780): -----------------
    07-31 14:09:42.088 I/Xposed  ( 7780): Added Xposed (/system/framework/XposedBridge.jar) to CLASSPATH
    07-31 14:09:42.218 I/Xposed  ( 7780): Detected ART runtime
    07-31 14:09:42.223 I/Xposed  ( 7780): Found Xposed class 'de/robv/android/xposed/XposedBridge', now initializing
    07-31 14:09:42.359 I/Xposed  ( 7780): Loading modules from /data/app/com.devadvance.rootcloak2-2/base.apk
    07-31 14:09:42.426 I/Xposed  ( 7780):   Loading class com.devadvance.rootcloak2.RootCloak
    07-31 14:10:09.642 W/Xposed  ( 8430): Package name for /data/app/com.google.android.GoogleCamera-1/base.apk had to be retrieved via parser


I'm afraid this wont help much because there doesnt seem to be any log about the app...

I enabled debug logs in rootcloak, deleted data from pivo and opened it again and then took log file from xposed

(Pivo force closes straight after pressing OK on the "access denied" message)

**Comment by hikaritenchi** [comment]
@voto123 There was a bug for debug logs, try again with the 20160731_2 beta and see if you can get more info

**Comment by voto123** [comment]
I had some problems with getting the log to show right in here (it was so big), I hope its okay that I link it to Mega...
It seems that now it made continuous log until xposed reached the log size limit.
Hopefully you can still find something out of it all.

[https://mega.nz/#!d5ZAQISR!7IoskhXO0Ecg6m8i9HYKBrXJcqFT7M8tHFahuumq3sA](Mega)

If you prefer some other hosting method for logs, let me know. I'm new in here, heh.

**Comment by hikaritenchi** [comment]
Try and get a normal log, not just the xposed messages. After reading some reviews of the app, it seems that Pivo might also be looking for apps such as Greenify as well as the normal "root" apps.

**Comment by voto123** [comment]
I couldnt get a proper log. I dont know if its possible to somehow filter unimportant logs out or if there's another way to get the logs...

But I did try to open the app after fresh install of android (RR-Remix ROM) with rootcloak set up.
Meaning that there shouldnt be any other apps which could block Pivo from working like greenify, amplify or others.
It still wouldnt work.

**Comment by ikoutone** [comment]
https://mega.nz/#!8tYXmRRb

Here is my Xposed log for Pivo app. Hopefully this gets solved somehow and we can use Pivo with rootcloak.

**Comment by iivmok** [comment]
Has there been any progress on this issue? I would love to use this app with root.

**Comment by mitjaimmonen** [comment]
@iivmok Pivo apparently needs Google's SafetyNet to pass for it to work. SafetyNet wont pass unless you use systemless root and Suhide for example. You can find Suhide on xda.

I tested it a month or so ago and had it working, so good luck if you are going to try!

**Comment by Mystard** [comment]
LineageOS 14.1 + Magisk + MagiskHide + Xposed + RootCloak = Pivo still complaining about developer options for me. This trash banking app is making me lose my mind...

**Comment by Juza89** [comment]
With OneplusOne i was finally able to get Pivo working with root! CyanogenOS 12.1.2 + Magisk + Magiskhide.
follow these intructions -> https://android.gadgethacks.com/how-to/magisk-101-switch-from-supersu-magisk-pass-safetynet-0177578/

**Comment by artoor32** [comment]
Finally worked with:
Fix development options and adb detect #179
